### PR TITLE
feat: Add azure mysql Db module

### LINKED ISSUE #649: Terratest Add Azure MySQL Module with Examples
This issue is a reference for contributions to Terratest for the Azure my sql module.

### PR DESCRIPTION
**Pull request by HadwaAbdelhalem**
Implements basic example for for the Azure mysql  Server and Database.

closes #649

### CODE REVIEW

File: examples/azure/terraform-azure-mysqldb-example/main.tf [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+resource "azurerm_resource_group" "mysql_rg" {
+  name     = "terratest-mysql-${var.postfix}"
+  location = var.location
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY AZURE MySQL SERVER
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "random_password" "password" {
```
Comment (yorinasub17) [inline examples/azure/terraform-azure-mysqldb-example/main.tf:30]: NIT: Add caveat about security in the same manner as https://github.com/gruntwork-io/terratest/blob/master/examples/azure/terraform-azure-vm-example/main.tf#L133

File: examples/azure/terraform-azure-mysqldb-example/main.tf [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  geo_redundant_backup_enabled      = false
+  infrastructure_encryption_enabled = true
+  backup_retention_days             = 7
+  public_network_access_enabled     = false
+  ssl_enforcement_enabled           = true
+  ssl_minimal_tls_version_enforced  = "TLS1_2"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY AZURE MySQL DATA BASE
```
Comment (yorinasub17) [inline examples/azure/terraform-azure-mysqldb-example/main.tf:58]:
```suggestion
# DEPLOY AZURE MySQL DATABASE
```

File: examples/azure/terraform-azure-mysqldb-example/variables.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    description = "The administrator login name for the mysql server."
+    type        = string
+    default     = "mysqladmin"
+}
+
+variable "mysqlserver_sku_name" {
+  description = "The SKU name for the mysql server."
+  type        = string
+  default     = "GP_Gen5_2"
+
```
Comment (yorinasub17) [inline examples/azure/terraform-azure-mysqldb-example/variables.tf:37]: NIT: extraneous whitespace

File: modules/azure/mysql.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+		return nil, err
+	}
+
+	// Attach authorizer to the client
+	mysqlClient.Authorizer = *authorizer
+
+	return &mysqlClient, nil
+}
+
+// GetMYSQLServer is a helper function that gets the server
```
Comment (yorinasub17) [inline modules/azure/mysql.go:34]:
```suggestion
// GetMYSQLServer is a helper function that gets the server.
```

File: modules/azure/mysql.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/mysql/mgmt/2017-12-01/mysql"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// GetMYSQLServerClient is a helper function that will setup a mysql server client
+func GetMYSQLServerClient(subscriptionID string) (*mysql.ServersClient, error) {
```
Comment (yorinasub17) [inline modules/azure/mysql.go:12]: Functions that return an error should be annotated with `E`.

File: modules/azure/mysql.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	//Return server
+	return &mysqlServer, nil
+}
+
+// GetMYSQLDBClient is a helper function that will setup a mysql DB client
+func GetMYSQLDBClient(subscriptionID string) (*mysql.DatabasesClient, error) {
```
Comment (yorinasub17) [inline modules/azure/mysql.go:62]: Functions that return an error should be annotated with `E`.

**Review by yorinasub17 — COMMENTED** [review]
Mostly LGTM! Had a few minor nits, mostly style based.

Can you post the link to the CI build once you make the updates? Thanks!

**Comment by HadwaAbdelhalem** [comment]
@yorinasub17 thanks Yori, Comments addressed., please find the CI execution here https://github.com/HadwaAbdelhalem/terratest/runs/1365646587?check_suite_focus=true

File: examples/azure/terraform-azure-mysqldb-example/main.tf [hunk 7]
```diff
@@ -27,6 +27,8 @@ resource "azurerm_resource_group" "mysql_rg" {
 # DEPLOY AZURE MySQL SERVER
 # ---------------------------------------------------------------------------------------------------------------------
 
+# Random password is used as an example to simplify the deployment and improve the security of the remote VM.
```
Comment (yorinasub17) [inline examples/azure/terraform-azure-mysqldb-example/main.tf:30]:
```suggestion
# Random password is used as an example to simplify the deployment and improve the security of the database.
```

**Review by yorinasub17 — COMMENTED** [review]
Thanks for the update! Found one typo. I can approve and merge this once that is fixed.

File: examples/azure/terraform-azure-mysqldb-example/main.tf (hunk 7 above)
Comment (HadwaAbdelhalem) [reply examples/azure/terraform-azure-mysqldb-example/main.tf]: fixed

**Review by yorinasub17 — APPROVED** [review]
LGTM!

**Pull request merged**